apps/accounts/models/provider_request.py

```diff
@@ -282,26 +286,51 @@ def approve(self) -> Hostingprovider:
                 hostingprovider=hp,
             )
 
+        # Fetch our archived documents:
+        # We want to compare the submitted evidence against these
+        # so we know which ones to make visible again, and which
+        # ones to leave archived
+        archived_documents = HostingProviderSupportingDocument.objects_all.filter(
+            archived=True, hostingprovider=hp
+        )
+        archived_doc_ids = []
+
+        def is_already_uploaded(evidence: ProviderRequestEvidence) -> Union[int, None]:
+            """
+            Check if the evidence is a content match for any of the archived documents
+            returning the if of the match if so
+            """
+            for doc in archived_documents:
+                logger.debug(evidence)
+                logger.debug(doc)
+                if evidence.has_content_match(doc):
+                    return doc.id
+            return None
+
         # create related objects: supporting documents
         for evidence in self.providerrequestevidence_set.all():
+            logger.debug(f"checking for matching evidence for: {evidence}")
+
             # AbstractSupportingDocument does not accept null values for `url`
             # and `attachment` fields
             url = evidence.link or ""
             attachment = evidence.file or ""
 
-            # check for the existence of archived evidence. Unarchive if found, like we do
```

Comment:
This is the key change - instead of running this check for every iteration of the loop, we fetch all the archived supporting documents once, and check the evidence against them, where the logic us defined in `is_already_uploaded` rather than the ORM query.

We do this because we don't have a reliable, fast way yet to compare the contents of the files from an ORM query.